docs/source/tutorials_source/package/tutorial_active_learning.py

```diff
@@ -0,0 +1,177 @@
+#    for uploading the embeddings and for defining the dataset for the classifier
+# C. Create a new dataset on the lightly platform as described in https://docs.lightly.ai/getting_started/platform.html#.
+#    Save the token and dataset id, you will need them later to upload the images and embeddings
+#    and to run the active learning samplers.
+# D. Upload the images to the platform, e.g. using the CLI https://docs.lightly.ai/getting_started/command_line_tool.html#upload-data-using-the-cli
+
+
+# %%
+# Imports
+# -------
+#
+# Import the Python frameworks we need for this tutorial.
+
+import csv
+from typing import List
+import numpy as np
+from sklearn.neighbors import KNeighborsClassifier
+
+from lightly.active_learning.agents.agent import ActiveLearningAgent
+from lightly.active_learning.config.sampler_config import SamplerConfig
+from lightly.active_learning.scorers.classification import ScorerClassification
+from lightly.api.api_workflow_client import ApiWorkflowClient
+from lightly.openapi_generated.swagger_client import SamplingMethod
+
+# %%
+# Definition of parameters
+path_to_embeddings_csv = "path/to/embeddings.csv"
+host = "https://api.lightly.ai"  # the URL of the web platform
+token = "ABCDEF"  # the token of the web platform
+dataset_id = "ASDFASDF"  # the id of your dataset on the web platform
+n_samples_to_sample = [100, 200]  # We first want to 100 samples to be labeled and
+# then another 100 to have 200 samples in total
+methods_to_sample = [SamplingMethod.CORESET, SamplingMethod.CORAL]  # We perform the initial sampling with CORESET
+# and then use CORAL for the following sampling with active learning scores
+
+
+# %%
+# Definition of a dataset for the classifier based on the embeddings.csv
+# The kNN-classifier needs the embeddings as features for its classification. Thus we define a class to create
+# such a dataset out of the embeddings.csv. It also allows to choose only a subset of all samples
+# dependent on the filenames given.
+
+class CSVEmbeddingDataset:
+    def __init__(self, path_to_embeddings_csv: str):
+        with open(path_to_embeddings_csv, 'r') as f:
+            data = csv.reader(f)
+
+            rows = list(data)
+            header_row = rows[0]
+            rows_without_header = rows[1:]
+
+            index_filenames = header_row.index('filenames')
+            filenames = [row[index_filenames] for row in rows_without_header]
+
+            index_labels = header_row.index('labels')
+            labels = [row[index_labels] for row in rows_without_header]
+
+            embeddings = rows_without_header
+            indexes_to_delete = sorted([index_filenames, index_labels], reverse=True)
+            for embedding_row in embeddings:
+                for index_to_delete in indexes_to_delete:
+                    del embedding_row[index_to_delete]
+
+        self.dataset = dict([(filename, (np.array(embedding_row, dtype=float), int(label)))
```

Comment:
could you add in comments below the rough structure of what we have now in `self.dataset`. This would help a lot for the understanding.